Source/WebCore/platform/network/NetworkStorageSession.cpp

```diff
@@ -216,14 +216,23 @@ ThirdPartyCookieBlockingDecision NetworkStorageSession::thirdPartyCookieBlocking
 
 bool NetworkStorageSession::shouldBlockCookies(const URL& firstPartyForCookies, const URL& resource, std::optional<FrameIdentifier> frameID, std::optional<PageIdentifier> pageID, ShouldRelaxThirdPartyCookieBlocking shouldRelaxThirdPartyCookieBlocking) const
 {
-    return thirdPartyCookieBlockingDecisionForRequest(firstPartyForCookies, resource, frameID, pageID, shouldRelaxThirdPartyCookieBlocking) == ThirdPartyCookieBlockingDecision::All;
+    return shouldBlockCookies(thirdPartyCookieBlockingDecisionForRequest(firstPartyForCookies, resource, frameID, pageID, shouldRelaxThirdPartyCookieBlocking));
 }
 
 bool NetworkStorageSession::shouldBlockCookies(const ResourceRequest& request, std::optional<FrameIdentifier> frameID, std::optional<PageIdentifier> pageID, ShouldRelaxThirdPartyCookieBlocking shouldRelaxThirdPartyCookieBlocking) const
 {
     return shouldBlockCookies(request.firstPartyForCookies(), request.url(), frameID, pageID, shouldRelaxThirdPartyCookieBlocking);
 }
 
+bool NetworkStorageSession::shouldBlockCookies(ThirdPartyCookieBlockingDecision thirdPartyCookieBlockingDecision) const
+{
+#if HAVE(ALLOW_ONLY_PARTITIONED_COOKIES)
+    if (!isOptInCookiePartitioningEnabled())
```

Comment:
I remember when `isOptInCookiePartitioningEnabled()` is false,  `thirdPartyCookieBlockingDecision` can only be `All` or `None`, so isn't `thirdPartyCookieBlockingDecision != WebCore::ThirdPartyCookieBlockingDecision::None` the same as `thirdPartyCookieBlockingDecision == ThirdPartyCookieBlockingDecision::All`?

Reply:
We don't currently check the flag here:
```
    switch (m_thirdPartyCookieBlockingMode) {
    case ThirdPartyCookieBlockingMode::All:
        return ThirdPartyCookieBlockingDecision::All;
 [...]
#if ENABLE(OPT_IN_PARTITIONED_COOKIES)
    case ThirdPartyCookieBlockingMode::AllExceptPartitioned:
        return ThirdPartyCookieBlockingDecision::AllExceptPartitioned;
#endif
```

You are recommending that we should check the flag in this last case and return `::All`? That's an alternative solution. You prefer that?

Reply:
Yes, that seems consistent with the existing implementation.